map_test.go

```diff
@@ -0,0 +1,518 @@
+//   Copyright 2015-2017 Ivan A Kostko (github.com/ivan-kostko; github.com/gopot)
+
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+
+//       http://www.apache.org/licenses/LICENSE-2.0
+
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package concurrentmap_test
+
+import (
+	. "github.com/gopot/concurrent-map"
+	"reflect"
+	"testing"
+)
+
+func TestNew(t *testing.T) {
+
+	testCases := []struct {
+		TestAlias    string
+		InitCapacity int
+		ExpectedMap  map[interface{}]interface{}
+	}{
+		{
+			TestAlias:    "Converting simple map into a concurrent map",
+			InitCapacity: 1,
+			ExpectedMap:  map[interface{}]interface{}{},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testAlias := testCase.TestAlias
+		initCapacity := testCase.InitCapacity
+		expectedMap := testCase.ExpectedMap
+
+		testFn := func(t *testing.T) {
+
+			actualCm := New(initCapacity)
+
+			if !(reflect.DeepEqual(actualCm.Items(), expectedMap)) {
```

Comment:
No, it the test of New(*) -> Items() cycle. There is nothing to Get(), Set has its own test cycle which includes New(*) as well as SetIfNotExists. So, from interface only Items() left to test.